CONTRIBUTING.md

```diff
@@ -12,20 +12,87 @@ As a developer:
+6. Push your changes to your master (or whatever you named your feature branch, that is entirely up to you at the end of the day)
+7. Initiate a pull request from your fork to the base repository
+   1. Remember: Opening a pull request is like saying "Hey maintainers, I have this change finalized and ready for you to spend time on reviewing it." The word `finalized` here is understood to imply that you are not planning on doing any more changes on the branch apart from when being asked to by the reviewers.
+   2. It is perfectly acceptable to open a pull request and mark it as `draft` (a GitHub feature) which then signals to the maintainers that if they have time, they are welcome to look at the change, but it may or may not be in its final form yet so you are not responsible for potential loss of time on their end if the review has to be performed multiple times on account of changes. Once you promote your draft PR to a real one, the comments from the point above apply however.
+   3. If your pull request contains a significant change, we recommend that you apply the similarly named github label on in it as well. It is okay if you do not do this, if we detect that the change is indeed significant, we will apply the label. If you do it in advance however, it will probably speed up the proceedings by removing one communication roundtrip from the review process of your pull request.
+8. Await CI, DCO & linting quality checks, as well as any feedback from reviewers
+9. If you need to update your pull requset either because you discovered an issue or because you were asked to do so we ask that you:
+   1.  try to add the change in a way that does not produce additional commits on the PR but instead do an `git commit --amend --signoff` on your local branch and then a force push to the remote branch of yours (the PR essentially). Again, if the change you are doing does not fit within any one of the existing commits of your PR, then it is justified to add a new commit and this is up to your discretion (maintainers may respectfully ask you to squash if they see otherwise)
+   2.  The rule of thumb for any and all things in git/Cactus is to maintain a clean, tidy commit log/history that enables everyone to easily look up changes and find accurate answers to the basic questions of `Who? / What? / When / Why?`. If you have ever been in a situation when you tried to figure out the original point a bug was introduced (and tried to figure out why the offending change was made in the first place) and the git blame just lead you to a 10 megabyte large patch with the message 'merge xyz', then you know exactly what it is we are trying to avoid here. :-)
+
```

Comment:
I think this is a great introductory guide. 
As a suggestion, which is potentially useful when the contributors base will be larger is to add some examples. For instance, in 5. i., explaining how to conduct such process. Or, alternatively, redirect to [sources like this one](https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history). 

Great work!